### PR TITLE
IGAPP-636: Fix tests and checks

### PR DESCRIPTION
**Pull request by steffenkleinle**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

File: native/src/routes/__tests__/TuNews.spec.tsx [hunk 1]
```diff
@@ -38,78 +48,77 @@ const news: [TunewsModel, TunewsModel] = [
...
+  it('should show language selector if language not available', async () => {
+    mocked(useLoadTuNews).mockImplementation(() => ({
+      ...tuNewsResponse,
+      availableLanguages: languages.slice(1),
+    }))
 
-    const { getByText, queryByText } = renderNews({ language: 'es' })
-    expect(getByText('languageNotAvailable')).toBeTruthy()
+    const { getByText, queryByText } = renderNews()
+    await waitFor(() => expect(getByText('languageNotAvailable')).toBeTruthy())
```
Comment (sarahsporck) [inline native/src/routes/__tests__/TuNews.spec.tsx:120]: findByText already wraps waitFor internally and should therefore also work here
```suggestion
    expect(await findByText('languageNotAvailable')).toBeTruthy()
```

**Review by sarahsporck — APPROVED** [review]
👍🏼 In `useLoadAsync.spec.tsx` you need to clear all mocks in before each. Everything else seems fine. Good job :)

**Review by SteffiStuffel — APPROVED** [review]
Looks good.

Wouldn't it be better to keep `react/jsx-no-constructed-context-values` rule active, but add useMemo everywhere. As this is a quite important feature, I would prefer the tests to check the production code as precise as possible.

**Comment by steffenkleinle** [comment]
> Looks good.
> 
> Wouldn't it be better to keep `react/jsx-no-constructed-context-values` rule active, but add useMemo everywhere. As this is a quite important feature, I would prefer the tests to check the production code as precise as possible.

Readded rule and fixed that in generally by moving the constructed object out of the rendering function. Was not possible in only one case where I disabled the rule manually. Using useMemo is usually not possible bc it is not a hook/react component.

**Pull request merged**